packages/webapp/src/components/Filter/FilterMultiSelectV2/index.tsx

```diff
@@ -0,0 +1,251 @@
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { MouseEvent, useEffect, useLayoutEffect, useMemo, useRef } from 'react';
+import Select, {
+  ClearIndicatorProps,
+  components,
+  MenuListProps,
+  MultiValue,
+  MultiValueRemoveProps,
+  OptionProps,
+} from 'react-select';
+import { ComponentFilter, ComponentFilterOption } from '../types';
+import { useState } from 'react';
+import { ReactComponent as XCircle } from '../../../assets/images/x-circle.svg';
+import { useTranslation } from 'react-i18next';
+import TextButton from '../../Form/Button/TextButton';
+import styles from './styles.module.scss';
+import Checkbox from '../../Form/Checkbox';
+import produce from 'immer';
+import { FilterItemProps } from '../FilterGroup';
+import { Label } from '../../Typography';
+
+interface FilterMultiSelectProps extends ComponentFilter {
+  onChange?: FilterItemProps['onChange'];
+  isDisabled?: boolean;
+}
+
+const MultiValueRemove = (props: MultiValueRemoveProps) => (
+  <components.MultiValueRemove {...props}>
+    <XCircle />
+  </components.MultiValueRemove>
+);
+
+const ClearIndicator = (props: ClearIndicatorProps<ComponentFilterOption>) => {
+  const { getValue, clearValue } = props;
+  const value = getValue();
+  const { t } = useTranslation();
+
+  const onClick = (e: MouseEvent) => {
+    e.stopPropagation();
+    e.preventDefault();
+    clearValue();
+  };
+
+  return (
+    <>
+      {t('FILTER.SELECTED', { count: value.length })} -
+      <TextButton className={styles.clearTextButton} onClick={onClick}>
+        clear
+      </TextButton>
+    </>
+  );
+};
+
+const Option = (props: OptionProps<ComponentFilterOption>) => {
+  const { label, isSelected, selectOption, data } = props;
+
+  const onClick = (e: MouseEvent) => {
+    selectOption(data);
+    e.stopPropagation();
+    e.preventDefault();
+  };
+
+  return (
+    <components.Option {...props}>
+      <TextButton className={styles.optionButton} onClick={onClick}>
+        <Checkbox label={label} checked={isSelected} />
+      </TextButton>
+    </components.Option>
+  );
+};
+
+const MenuList = (props: MenuListProps<ComponentFilterOption>) => {
+  const { children, getValue, hasValue, options, clearValue } = props;
+  const { t } = useTranslation();
+
+  const partiallyChecked = getValue().length !== options.length;
+
+  return (
+    <components.MenuList {...props}>
+      <div className={styles.option}>
```

Comment:
Both the checked and partially checked states need to have the [selected option styling](https://www.figma.com/file/eMffC0h3JPlWtzNttxJpil/Litefarm-Updated?type=design&node-id=2805-37627&mode=design&t=mloVzFJhA0P2XRXV-0) (the grey background).